### PR TITLE
Use better type safety when creating URLs

### PR DESCRIPTION
**Pull request by acoburn**
# Improve type safety of URLs generated for a client ID document

## Description of changes

At present, the URLs used in a rendered client ID document use string concatenation based on a client-supplied header, which could open the door to malicious behavior, especially if the redirect URLs are manipulated. 

## Commit checklist

- [ ] All acceptance criteria are met.
- [ ] Includes tests to ensure functionality, accessibility and prevent regressions, including E2E tests for supported browsers.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [ ] A changeset version has been created, if applicable.
- [ ] Meets Inrupt coding standards and adheres to commit conventions.

## Design requirements checklist

- [ ] Meets Inrupt API Design and UX standards
- [ ] Is responsive to our documented minimum supported screen size + resolution
- [ ] Code is extensible by external contributors or anyone forking

### CODE REVIEW

File: src/helpers/clientId/clientId.ts [hunk 1]
```diff
@@ -21,18 +21,18 @@
...
     // URLs the user will be redirected back to upon successful authentication:
-    redirect_uris: [hostname, hostname.concat("login")],
+    redirect_uris: [hostname, new URL("login", hostname)],
     // URLs the user can be redirected to back to upon successful logout:
     post_logout_redirect_uris: [
       hostname,
-      hostname.concat("login"),
-      hostname.concat("*"),
+      new URL("login", hostname),
+      new URL("*", hostname),
```
Comment (acoburn) [inline src/helpers/clientId/clientId.ts:35]: Note that this is not a valid `post_logout_redirect_uri`

Comment (NSeydoux) [reply src/helpers/clientId/clientId.ts:35]: Let's remove it from the client ID document. AMC is meant to be a reference implementation, this client ID document is going to be used as an example by some developers.

**Review by NSeydoux — APPROVED** [review]
Looks good. The preview fails on login, but that is because of Vercel. More things happen in the preview than I realized: if you `curl https://authorization-management-component-jzsxmwmmk-inrupt.vercel.app/api/app`, you get a result very different from what I anticipated.

File: src/helpers/clientId/clientId.ts (hunk 1 above)
Comment (acoburn) [reply src/helpers/clientId/clientId.ts]: Removed in https://github.com/inrupt/authorization-management-component/pull/36/commits/da2e1df4703573cf07feb0b8b1de6f2eddc9622e

**Pull request merged**